sormas-ui/src/main/java/de/symeda/sormas/ui/task/BulkTaskDataForm.java

```diff
@@ -0,0 +1,169 @@
+		super(TaskBulkEditData.class, TaskDto.I18N_PREFIX);
+		this.district = district;
+		setWidth(680, Unit.PIXELS);
+		hideValidationUntilNextCommit();
+		initialized = true;
+		addFields();
+	}
+
+	@Override
+	protected void addFields() {
+
+		if (!initialized) {
+			return;
+		}
+
+		taskStatusCheckbox = new CheckBox(I18nProperties.getCaption(Captions.bulkTaskStatus));
+		getContent().addComponent(taskStatusCheckbox, STATUS_CHECKBOX);
+		NullableOptionGroup status = addField(TaskBulkEditData.TASK_STATUS, NullableOptionGroup.class);
+		status.setEnabled(false);
+		FieldHelper.setRequiredWhen(getFieldGroup(), taskStatusCheckbox, Arrays.asList(TaskBulkEditData.TASK_STATUS), Arrays.asList(true));
+		taskStatusCheckbox.addValueChangeListener(e -> {
+			status.setEnabled((boolean) e.getProperty().getValue());
+		});
+
+		priorityCheckbox = new CheckBox(I18nProperties.getCaption(Captions.bulkTaskPriority));
+		getContent().addComponent(priorityCheckbox, PRORITY_CHECKBOX);
+		NullableOptionGroup priority = addField(TaskBulkEditData.TASK_PRIORITY, NullableOptionGroup.class);
+		priority.setEnabled(false);
+		FieldHelper.setRequiredWhen(getFieldGroup(), priorityCheckbox, Arrays.asList(TaskBulkEditData.TASK_PRIORITY), Arrays.asList(true));
+		priorityCheckbox.addValueChangeListener(e -> {
+			priority.setEnabled((boolean) e.getProperty().getValue());
+		});
+
+		if (district != null) {
+			assigneeCheckbox = new CheckBox(I18nProperties.getCaption(Captions.bulkTaskassignee));
+			getContent().addComponent(assigneeCheckbox, ASSIGNEE_CHECKBOX);
+			ComboBox assignee = addField(TaskBulkEditData.TASK_ASSIGNEE, ComboBox.class);
+			assignee.setEnabled(false);
+			FieldHelper.addSoftRequiredStyleWhen(
+					getFieldGroup(),
+					assigneeCheckbox,
+					Arrays.asList(TaskBulkEditData.TASK_ASSIGNEE),
+					Arrays.asList(true),
+					null);
+
+			List<UserReferenceDto> users = getUsers();
+			Map<String, Long> userTaskCounts =
+					FacadeProvider.getTaskFacade().getPendingTaskCountPerUser(users.stream().map(ReferenceDto::getUuid).collect(Collectors.toList()));
+			for (UserReferenceDto user : users) {
+				assignee.addItem(user);
+				Long userTaskCount = userTaskCounts.get(user.getUuid());
+				assignee.setItemCaption(user, user.getCaption() + " (" + (userTaskCount != null ? userTaskCount.toString() : "0") + ")");
+			}
+
+			assigneeCheckbox.addValueChangeListener(e -> {
+				assignee.setEnabled((boolean) e.getProperty().getValue());
+			});
+		}
+	}
+
+	private List<UserReferenceDto> getUsers() {
+		final List<UserReferenceDto> users = new ArrayList<>();
+		UserDto userDto = UserProvider.getCurrent().getUser();
+
+		if (district != null) {
+			users.addAll(FacadeProvider.getUserFacade().getUserRefsByDistrict(district, true));
+		}
+
+		// Allow regional users to assign the task to national ones
+		if (FacadeProvider.getFeatureConfigurationFacade().isFeatureEnabled(FeatureType.ASSIGN_TASKS_TO_HIGHER_LEVEL)
```

Comment:
#4582 has just introduced an extended logic that allows users of every user role to assign tasks to users of the next higher jurisdiction level. Please have a look at the changed code and make sure that the code here works identically

Reply:
This part of the code now has to be removed (lines 154 to 161); the generic code that you added makes it redundant